website/content/docs/pipelines/index.mdx

```diff
@@ -0,0 +1,438 @@
+----+----+-----------+---------+-------------+-------------+----------------------------
+  ✔ | 18 | default   | pack    | 3 hours ago | 3 hours ago | test[run: 2][step: build]
+  ✔ | 17 | default   | pack    | 4 hours ago | 4 hours ago | test[run: 1][step: build]
+```
+
+### Exec Step Plugin
+
+Pipelines come with a built-in step plugin named `exec`. This plugin allows a
+step to execute bash within the Pipeline run. This could be used to run tests
+after a build or deploy, or use tools packaged into the exec step image to
+security scan the built artifact.
+
+```hcl
+pipeline "example" {
+  step "run-some-code" {
+    image_url = "https://localhost:5000/waypoint-odr:dev"
+
+    use "exec" {
+      # executes a binary test with some arguments
+      command = "test"
+      args    = ["--full", "test-all"]
+    }
+  }
+}
+```
+
+### Embedded Pipelines in Steps
+
+You can optionally define a full pipeline within a Pipeline Step. This pattern
+is useful if you have a series of steps that make sense to be grouped together,
+such as doing a deploy and release within a specific workspace.
+
+```hcl
+pipeline "simple-nested" {
+  step "build" {
+    use "build" {}
+  }
+
+  step "deploy" {
+    pipeline "deploy" {
+      step "deploy" {
+        use "deploy" {
+        }
+      }
+      step "release" {
+        use "release" {
+          prune = true
+        }
+      }
+    }
+  }
+}
+```
+
+~> **NOTE**: Currently, this Tech Preview only allows for 1 level of nesting when defining an embedded pipeline. However, Pipelines can also be used as a reference within a step.
+
+### Reference Pipelines As Steps
+
+You can reference other Pipelines within a Project via a `pipeline` step:
+
+```hcl
+project = "my-pipeline-project"
+
+pipeline "simple-nested" {
+  step "build" {
+    use "build" {}
+  }
+
+  step "my-deploy" {
+    #  "pipeline" is a special "built-in" step that references another pipeline
+    use "pipeline" {
+      project = "my-pipeline-project"
+      name    = "deploy"
+    }
+  }
+}
+
+pipeline "deploy" {
+  step "deploy" {
+    use "deploy" {}
+  }
+  step "release" {
+    use "release" {
+      prune = true
+    }
+  }
+}
+```
+
+### Pipelines Across Workspaces
+
+Pipelines can be defined and scoped to a Workspace for deploying in various environments.
+When you use the pipeline run CLI, you can run a pipeline in a different workspace
+with the `-workspace` flag.
+
+```shell
+$ workspace pipeline run builder -workspace=dev
+```
+
+#### Workspace Scoped Steps
+
+You can override the environment that a Pipeline step runs in by specifying a
+`workspace` param for any step in your Pipeline definition. For example,
+this pipelines "build-prod" step will run in the production workspace.
+
+Notice that in the app definition, it contains two build stanzas, one of which
+is _workspace scoped_. That means when the "build-prod" step runs, it will
+use that workspace scoped build plugin pack rather than the default "docker" plugin.
```

Comment:
```suggestion
Notice that the "build" stanza in the "app" definition contains two "use" plugin definitions, 
one of which is _workspace scoped_. That means when the "build-prod" step runs, it will
use the "prod" workspace scoped build plugin pack rather than the default "docker" plugin.
```